vsphere_server_plugin/server.py

```diff
@@ -1125,7 +1159,132 @@ def update(server_client, **_):
+                nicspec.device = nic
+                nicspec.operation = \
+                    vim.vm.device.VirtualDeviceSpec.Operation.remove
+                device_changes.append(nicspec)
+
+        datacenter_name = server_client.cfg['datacenter_name']
+        datacenter = server_client._get_obj_by_name(vim.Datacenter,
+                                                    datacenter_name)
+        # go through the new networks to add them
+        for network in new_networks:
+            if network.get('name') in to_add:
+                network['name'] = \
+                    server_client._get_connected_network_name(network)
+                nicspec, _ = \
+                    server_client._add_network(network, datacenter, netcnt)
+                device_changes.append(nicspec)
+                netcnt += 1
+
+        if device_changes:
+            vmconf = vim.vm.ConfigSpec()
+            vmconf.deviceChange = device_changes
+            task = server_obj.obj.ReconfigVM_Task(spec=vmconf)
+            server_client._wait_for_task(task)
+
+        # get server object again to update networks
+        server_obj = \
+            server_client._get_obj_by_id(
+                vim.VirtualMachine,
+                ctx.instance.runtime_properties[VSPHERE_SERVER_ID],
+                use_cache=False)
+        store_server_details(server_client, server_obj)
+
+        # update expected configuration after update
+        expected_configuration = {}
+        network = json.loads(json.dumps(server_obj.network,
+                                        cls=VmomiSupport.VmomiJSONEncoder,
+                                        sort_keys=True, indent=4))
+        summary = json.loads(json.dumps(server_obj.summary.config,
+                                        cls=VmomiSupport.VmomiJSONEncoder,
+                                        sort_keys=True, indent=4))
+        expected_configuration['network'] = network
+        expected_configuration['summary'] = summary
+
+        ctx.instance.runtime_properties[
+            'expected_configuration'] = expected_configuration
+
+        if new_networks:
+            # fetch new IP
+            default_ip = None
+            manager_network_ip = None
+            management_networks = \
+                [network['name'] for network
+                 in networking.get('connect_networks', [])
+                 if network.get('management', False)]
+            management_network_name = (management_networks[0]
+                                       if len(management_networks) == 1
+                                       else None)
+            # make sure that networks have IPs
+            while not server_obj.guest.net:
+                server_obj = \
+                    server_client._get_obj_by_id(
+                        vim.VirtualMachine,
+                        ctx.instance.runtime_properties[VSPHERE_SERVER_ID],
+                        use_cache=False)
+                time.sleep(10)
```

Comment:
Is 10 a magic number? Would it be better to do a retry or to poll this?

Reply:
not really a magic number just adding small delay for the vm-tools to get new IP [ usually it is more than enough, but to be safe I made it 10 seconds]